### PR TITLE
0.6 e2e test cases

### PR DESCRIPTION
**Pull request by sakows**

### CODE REVIEW

File: istio/tests/cases/3_route_all_v3_spec.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+with description('nginmesh Test 03'):
+    with before.all:
+         #Read Config file
+         configuration.setenv(self)
+
+    with context('Set environment'):
+         with it('Bookinfo add Routing Rule'):
+            Rule.add(rule_name)
+            time.sleep(5)
```
Comment (sehz) [inline istio/tests/cases/3_route_all_v3_spec.py]: this is fragile.  instead, confirm rules have successfully applied

File: istio/tests/cases/4_route_reviews_v1_50_v3_spec.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+with description('nginmesh Test 04'):
+    with before.all:
+         #Read Config file
+         configuration.setenv(self)
+
+    with context('Set environment'):
+         with it('Bookinfo add Routing Rule'):
+            Rule.add(rule_name)
+            time.sleep(5)
```
Comment (sehz) [inline istio/tests/cases/4_route_reviews_v1_50_v3_spec.py]: same here

File: istio/tests/cases/5_route_reviews_v1_75_v2_25_spec.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+with description('nginmesh Test 05'):
+    with before.all:
+         #Read Config file
+         configuration.setenv(self)
+
+    with context('Set environment'):
+         with it('Bookinfo add Routing Rule'):
+            Rule.add(rule_name)
+            time.sleep(10)
```
Comment (sehz) [inline istio/tests/cases/5_route_reviews_v1_75_v2_25_spec.py]: same as above

File: istio/tests/cases/5_route_reviews_v1_75_v2_25_spec.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    with context('Starting Test'):
+        with it('Bookinfo destination-weight test, route to V1-75%, V2-25%'):
+            configuration.generate_request(self)
+            expect(self.v1_count).not_to(equal(0))
+            expect(self.v2_count).not_to(equal(0))
+            expect(self.v3_count).to(equal(0))
+            expect(self.v1_count>self.v2_count).to(be_true)
+            if self.performance=='on':
```
Comment (sehz) [inline istio/tests/cases/5_route_reviews_v1_75_v2_25_spec.py]: wrecker config is repetive. can you abstract it out?

File: istio/tests/cases/5_route_reviews_v1_75_v2_25_spec.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                print performance.wrecker(self.GATEWAY_URL)
+            else:
+                pass
+
+    with context('Clean Environment'):
```
Comment (sehz) [inline istio/tests/cases/5_route_reviews_v1_75_v2_25_spec.py]: this is duplicate.  maybe create super class to wrap this.

File: istio/tests/cases/6_route_reviews_v2_50_v3_spec.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+with description('nginmesh Test 06'):
+    with before.all:
+         #Read Config file
+         configuration.setenv(self)
+
+    with context('Set environment'):
+         with it('Bookinfo add Routing Rule'):
+            Rule.add(rule_name)
+            time.sleep(5)
```
Comment (sehz) [inline istio/tests/cases/6_route_reviews_v2_50_v3_spec.py]: same as above

File: istio/tests/cases/configuration.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    self.deploy_bookinfo_app='on'
+    self.v1_count=0
+    self.v2_count=0
+    self.v3_count=0
+    self.total_count = 0
+    return self.performance,self.GATEWAY_URL,self.v1_count,self.v2_count,self.v3_count,self.total_count,self.VERSION
+
+class Istio:
+    def install_istio(self):
+         subprocess.call("kubectl apply -f ../../release/samples/bookinfo/kube/istio.yaml > /dev/null 2>&1 | exit 0",universal_newlines=True,shell=True)
```
Comment (sehz) [inline istio/tests/cases/configuration.py]: put istio path as variable instead of duplicating

File: istio/tests/cases/configuration.py [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+           self.v3_count+=1
+        else:
+           self.total_count += 1
+    print(" | V1 Hit="+str(self.v1_count)+" | V2 Hit="+str(self.v2_count)+" | V3 Hit="+str(self.v3_count)+" | Total Hit="+str(self.total_count)+ " |")
+    return self.GATEWAY_URL,self.v1_count,self.v2_count,self.v3_count,self.total_count,self.VERSION, self.performance
+
+
+class Rule:
+     def add(self,rule_name):
+         subprocess.call("kubectl create -f ../../release/samples/bookinfo/kube/"+rule_name+" > /dev/null 2>&1 | exit 0",universal_newlines=True,shell=True)
```
Comment (sehz) [inline istio/tests/cases/configuration.py]: same here. don't duplicate path

File: istio/tests/src/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,37 @@
+FROM ubuntu:16.04
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends -y build-essential  \
+      git curl ca-certificates python python-setuptools python-pip libssl-dev nano curl
+# install github.com/wg/wrk v3.1.0
+RUN (mkdir /opt/wrk && cd /opt/wrk && curl -L# https://github.com/wg/wrk/archive/3.1.0.tar.gz | tar zx --strip 1 && make && mv wrk /bin)
```
Comment (sehz) [inline istio/tests/src/Dockerfile]: make wrecker version configurable using Docker Arg

File: istio/tests/src/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+COPY . /bookinfo_spec
+ARG VCS_REF
+ARG BUILD_DATE
+# Metadata
+LABEL org.label-schema.vcs-ref=$VCS_REF \
+      org.label-schema.vcs-url="https://github.com/lachie83/k8s-kubectl" \
+      org.label-schema.build-date=$BUILD_DATE \
+      org.label-schema.docker.dockerfile="/Dockerfile"
+ENV KUBE_LATEST_VERSION="v1.9.4"
+ADD https://storage.googleapis.com/kubernetes-release/release/v1.9.1/bin/linux/amd64/kubectl /usr/local/bin/kubectl
```
Comment (sehz) [inline istio/tests/src/Dockerfile]: make kubernetes version configurable

File: istio/tests/src/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+ADD  https://git.io/getLatestIstio  /bookinfo_spec
+# RUN export PATH=istio-0.3.0/bin:$PATH
+RUN which python
+ADD  https://github.com/nginmesh/nginmesh/releases/download/0.6.0/nginmesh-0.6.0.tar.gz /bookinfo_spec
```
Comment (sehz) [inline istio/tests/src/Dockerfile]: make nginmesh version configurable using ARG

File: istio/tests/cases/install.sh [hunk 12]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+: '
+os=`uname`
+if [[ "$os" == 'Darwin' ]]; then
```
Comment (sehz) [inline istio/tests/cases/install.sh]: don't install wrk, kubectl.  just put as part of installation in README.

File: istio/tests/cases/install.sh [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+    echo "macOS is detected."
+
+    python --version && pip -v > /dev/null 2>&1
+    if [ ! $? -eq 0 ]; then
+    echo "install Python and pip"
+    brew install python2 && easy_install pip
```
Comment (sehz) [inline istio/tests/cases/install.sh]: don't install python, pip.  put as part of instruction

Comment (sehz) [inline istio/tests/cases/install.sh]: mamba doc recommend use of pipenv

**Review by sehz — CHANGES_REQUESTED** [review]

File: istio/tests/cases/install.sh (hunk 13 above)
Comment (sakows) [reply istio/tests/cases/install.sh]: ok

Comment (sakows) [reply istio/tests/cases/install.sh]: ok

File: istio/tests/cases/install.sh (hunk 12 above)
Comment (sakows) [reply istio/tests/cases/install.sh]: ok

File: istio/tests/src/Dockerfile (hunk 11 above)
Comment (sakows) [reply istio/tests/src/Dockerfile]: ok

File: istio/tests/src/Dockerfile (hunk 10 above)
Comment (sakows) [reply istio/tests/src/Dockerfile]: ok

File: istio/tests/cases/5_route_reviews_v1_75_v2_25_spec.py (hunk 3 above)
Comment (sakows) [reply istio/tests/cases/5_route_reviews_v1_75_v2_25_spec.py]: ok

File: istio/tests/cases/4_route_reviews_v1_50_v3_spec.py (hunk 2 above)
Comment (sakows) [reply istio/tests/cases/4_route_reviews_v1_50_v3_spec.py]: ok

File: istio/tests/cases/3_route_all_v3_spec.py (hunk 1 above)
Comment (sakows) [reply istio/tests/cases/3_route_all_v3_spec.py]: ok

File: istio/agent/vendor/golang.org/x/sys/unix/linux/Dockerfile [hunk 14]
```diff
@@ -29,7 +29,7 @@ RUN curl -fsSL "$GOLANG_DOWNLOAD_URL" -o golang.tar.gz \
 
 ENV PATH /usr/local/go/bin:$PATH
 
-# Linux and Glibc build dependencies
+# Linux and Glibc src dependencies
 RUN apt-get update && apt-get install -y \
```
Comment (sehz) [inline istio/agent/vendor/golang.org/x/sys/unix/linux/Dockerfile]: why change in this file?

File: istio/agent/vendor/golang.org/x/sys/unix/mkerrors.sh [hunk 15]
```diff
@@ -471,7 +471,7 @@ echo '#include <signal.h>' | $CC -x c - -E -dM $ccflags |
 echo '// mkerrors.sh' "$@"
 echo '// Code generated by the command above; see README.md. DO NOT EDIT.'
```
Comment (sehz) [inline istio/agent/vendor/golang.org/x/sys/unix/mkerrors.sh]: what's reason for change?

File: istio/tests/Makefile [hunk 16]
```diff
@@ -0,0 +1,8 @@
+tag = gcr.io/nginsako/bookinfo_testing:latest
+
+build:
```
Comment (sehz) [inline istio/tests/Makefile]: put version as variable

File: istio/tests/cases/README.md [hunk 17]
```diff
@@ -0,0 +1,170 @@
...
+
+### Run Test 
+1. Change directory to /cases directory in nginMesh repo:
+```
+cd tests/cases
+```
+2. Install python dependencies in virtual environment:
+```
+./install.sh
+```
```
Comment (sehz) [inline istio/tests/cases/README.md]: update the README to use virtual environment.

File: istio/tests/cases/configuration.py [hunk 18]
```diff
@@ -0,0 +1,72 @@
+import subprocess
+import grequests
+import performance
+import time
+VERSION='0.6.0'
+rule_apply_time=5
+# istio_path="../../release/samples/bookinfo/kube/"
+istio_path="nginmesh-0.6.0/samples/bookinfo/kube/"
```
Comment (sehz) [inline istio/tests/cases/configuration.py]: this is not istio path. it is nginmesh path

Comment (sehz) [reply istio/tests/cases/configuration.py]: refer to local source path not release

File: istio/tests/cases/configuration.py [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+
+def setenv(self):
+    self.GATEWAY_URL = str(subprocess.check_output("kubectl get svc -n istio-system | grep -E 'istio-ingress' | awk '{ print $4 }'", universal_newlines=True,shell=True)).rstrip()
+    self.url = "http://"+self.GATEWAY_URL+"/productpage"
+   # self.zipkin="http://localhost:9412/api/v2/services"
+   # self.prometheus="http://localhost:9090/api/v1/query?query=http_requests_total"
+   # self.servicegraph="http://localhost:8088/graph"
+   # self.grafana="http://localhost:3000/api/dashboards/db/istio-dashboard"
+    self.VERSION='0.6.0'
```
Comment (sehz) [inline istio/tests/cases/configuration.py]: put version as environment variable

File: istio/agent/vendor/golang.org/x/sys/unix/linux/Dockerfile (hunk 14 above)
Comment (sakows) [reply istio/agent/vendor/golang.org/x/sys/unix/linux/Dockerfile]: during rename IDE updated it as well, will rollback.

File: istio/agent/vendor/golang.org/x/sys/unix/mkerrors.sh (hunk 15 above)
Comment (sakows) [reply istio/agent/vendor/golang.org/x/sys/unix/mkerrors.sh]: during rename IDE updated it as well, will rollback.

Comment (sakows) [reply istio/agent/vendor/golang.org/x/sys/unix/mkerrors.sh]: during rename IDE updated it as well, will rollback.

File: istio/tests/cases/configuration.py (hunk 18 above)
Comment (sakows) [reply istio/tests/cases/configuration.py]: ok

File: istio/tests/cases/README.md (hunk 17 above)
Comment (sakows) [reply istio/tests/cases/README.md]: ok

File: istio/tests/cases/configuration.py (hunk 19 above)
Comment (sakows) [reply istio/tests/cases/configuration.py]: refactored, we dont need Version variable anymore.

File: istio/tests/Makefile [hunk 20]
```diff
@@ -0,0 +1,27 @@
+KUBE_VERSION=v1.9.4
+NGINMESH_VERSION=0.6.0
+WRK_VERSION=3.1.0
+GIT_COMMIT=$(shell git rev-parse --short HEAD)
+TEST_IMAGE_NAME=bookinfo_testing
+REPO=gcr.io/$(GCLOUD_PROJECT)
+TAG=-$(NGINMESH_VERSION)-$(GIT_COMMIT)
+TAG_LATEST=$(NGINMESH_VERSION)
+DOCKER_PUSH = gcloud docker -- push
+
```
Comment (sehz) [inline istio/tests/Makefile]: separate into release and dev.  release should use nginmesh namespace.  gcp is only for dev.

File: istio/tests/Makefile [hunk 21]
```diff
@@ -0,0 +1,27 @@
...
+build:
+	docker build --build-arg KUBE_VERSION=$(KUBE_VERSION)  --build-arg NGINMESH_VERSION=$(NGINMESH_VERSION)  --build-arg WRK_VERSION=$(WRK_VERSION) -t $(REPO)/$(TEST_IMAGE_NAME):$(TAG_LATEST) -f src/Dockerfile .
+
+build-clean:
```
Comment (sehz) [inline istio/tests/Makefile]: what's purpose of build-clean? this is probably not needed

File: istio/tests/Makefile [hunk 22]
```diff
@@ -0,0 +1,27 @@
...
+	docker build --no-cache --build-arg KUBE_VERSION=$(KUBE_VERSION)  --build-arg NGINMESH_VERSION=$(NGINMESH_VERSION)  --build-arg WRK_VERSION=$(WRK_VERSION) -t $(REPO)/$(TEST_IMAGE_NAME):$(TAG_LATEST) -f src/Dockerfile .
+
+push-gcp:
+	$(DOCKER_PUSH)  $(REPO)/$(TEST_IMAGE_NAME):$(TAG_LATEST)
+
+clean-docker:
```
Comment (sehz) [inline istio/tests/Makefile]: these 2 are not probably needed also

File: istio/tests/cases/configuration.py [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+    self.total_count = 0
+    return self.performance,self.GATEWAY_URL,self.v1_count,self.v2_count,self.v3_count,self.total_count
+
+def generate_request(self, rule_name=None):
+    self.v1_count=0
+    self.v2_count=0
+    self.v3_count=0
+    self.total_count = 0
+    if rule_name !="route-rule-reviews-test-v2.yaml" and rule_name !="route-rule-http-redirect.yaml" and rule_name !="route-rule-http-retry.yaml" :
+        urls = [self.url for i in range(10)]
```
Comment (sehz) [inline istio/tests/cases/configuration.py]: don't hard code number.

File: istio/tests/cases/configuration.py [hunk 24]
```diff
@@ -0,0 +1,66 @@
...
+    else:
+        pass
+
+    return self.GATEWAY_URL,self.v1_count,self.v2_count,self.v3_count,self.total_count, self.performance
+
+
+class Rule:
+     def add(self,rule_name):
+         subprocess.call("kubectl create -f "+nginmesh_rule_path+rule_name+" > /dev/null 2>&1 | exit 0",universal_newlines=True,shell=True)
+         time.sleep(rule_apply_time)
```
Comment (sehz) [inline istio/tests/cases/configuration.py]: create helper for deploying rule with sleep, that way remove duplicate code

File: istio/tests/cases/configuration.py [hunk 25]
```diff
@@ -0,0 +1,66 @@
...
+
+     def delete(self,rule_name):
+         subprocess.call("kubectl delete -f "+nginmesh_rule_path+rule_name+" > /dev/null 2>&1 | exit 0",universal_newlines=True,shell=True)
```
Comment (sehz) [inline istio/tests/cases/configuration.py]: create helper for delete as well

File: istio/tests/cases/configuration.py (hunk 24 above)
Comment (sehz) [reply istio/tests/cases/configuration.py]: might want to use python K8 API:  https://github.com/kubernetes-client/python

File: istio/tests/src/Dockerfile [hunk 26]
```diff
@@ -0,0 +1,33 @@
...
+
+# ADD Istio
+# ADD  https://git.io/getLatestIstio  /bookinfo_spec
+
+# ADD dependencies
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends -y build-essential  \
+      git curl ca-certificates python python-setuptools python-pip libssl-dev nano
+# ADD nginMesh
+RUN  curl -L# https://github.com/nginmesh/nginmesh/releases/download/$NGINMESH_VERSION/nginmesh-$NGINMESH_VERSION.tar.gz  | tar zx
```
Comment (sehyoc) [inline istio/tests/src/Dockerfile]: should install from source code not from release.

File: istio/tests/src/bookinfo-test.yaml [hunk 27]
```diff
@@ -0,0 +1,49 @@
+##################################################################################################
+#  Bookinfo-Test
```
Comment (sehz) [inline istio/tests/src/bookinfo-test.yaml:2]: what is this for?  this is also uses your own image. won't work for anyone else....

File: istio/tests/Makefile (hunk 21 above)
Comment (sakows) [reply istio/tests/Makefile]: ok, it is for rebuild image without using  cache

File: istio/tests/cases/configuration.py (hunk 23 above)
Comment (sakows) [reply istio/tests/cases/configuration.py]: ok

File: istio/tests/Makefile (hunk 22 above)
Comment (sakows) [reply istio/tests/Makefile]: ok

File: istio/tests/cases/configuration.py (hunk 24 above)
Comment (sakows) [reply istio/tests/cases/configuration.py]: ok

File: istio/tests/cases/configuration.py (hunk 25 above)
Comment (sakows) [reply istio/tests/cases/configuration.py]: ok

File: istio/tests/src/bookinfo-test.yaml (hunk 27 above)
Comment (sakows) [reply istio/tests/src/bookinfo-test.yaml:2]: ok

File: istio/tests/Makefile (hunk 20 above)
Comment (sakows) [reply istio/tests/Makefile]: ok

File: istio/release/samples/bookinfo/kube/route-rule-all-v1.yaml [hunk 28]
```diff
@@ -2,6 +2,7 @@ apiVersion: config.istio.io/v1alpha2
 kind: RouteRule
 metadata:
   name: productpage-default
+  namespace: default
```
Comment (sehz) [inline istio/release/samples/bookinfo/kube/route-rule-all-v1.yaml]: revert these changes.   the rule file should not be changed

File: istio/release/samples/bookinfo/kube/route-rule-http-redirect.yaml [hunk 29]
```diff
@@ -1,7 +1,8 @@
-apiVersion: config.istio.io/v1alpha3
```
Comment (sehz) [inline istio/release/samples/bookinfo/kube/route-rule-http-redirect.yaml:1]: is this old rule? I dont' see in Istio release. if so please delete this

File: istio/release/samples/bookinfo/kube/route-rule-http-retry.yaml [hunk 30]
```diff
@@ -2,6 +2,7 @@ apiVersion: config.istio.io/v1alpha2
 kind: RouteRule
 metadata:
```
Comment (sehz) [inline istio/release/samples/bookinfo/kube/route-rule-http-retry.yaml]: don't see in this 0.6 release

File: istio/release/samples/bookinfo/kube/route-rule-reviews-50-v3.yaml [hunk 31]
```diff
@@ -2,6 +2,7 @@ apiVersion: config.istio.io/v1alpha2
 kind: RouteRule
 metadata:
   name: reviews-default
+  namespace: default
```
Comment (sehz) [inline istio/release/samples/bookinfo/kube/route-rule-reviews-50-v3.yaml]: revert this

File: istio/release/samples/bookinfo/kube/route-rule-reviews-test-v2.yaml [hunk 32]
```diff
@@ -2,6 +2,7 @@ apiVersion: config.istio.io/v1alpha2
 kind: RouteRule
 metadata:
   name: reviews-test-v2
+  namespace: default
```
Comment (sehz) [inline istio/release/samples/bookinfo/kube/route-rule-reviews-test-v2.yaml]: revert this

File: istio/release/samples/bookinfo/kube/route-rule-reviews-v2-v3.yaml [hunk 33]
```diff
@@ -2,6 +2,7 @@ apiVersion: config.istio.io/v1alpha2
 kind: RouteRule
 metadata:
   name: reviews-default
+  namespace: default
```
Comment (sehz) [inline istio/release/samples/bookinfo/kube/route-rule-reviews-v2-v3.yaml]: revert this

File: istio/release/samples/bookinfo/kube/route-rule-reviews-v3.yaml [hunk 34]
```diff
@@ -2,6 +2,7 @@ apiVersion: config.istio.io/v1alpha2
 kind: RouteRule
 metadata:
   name: reviews-default
+  namespace: default
```
Comment (sehz) [inline istio/release/samples/bookinfo/kube/route-rule-reviews-v3.yaml]: revert this

File: istio/tests/Makefile [hunk 35]
```diff
@@ -0,0 +1,17 @@
...
+NGINMESH_VERSION=0.6.0
+WRK_VERSION=3.1.0
+GIT_COMMIT=$(shell git rev-parse --short HEAD)
+TEST_IMAGE_NAME=bookinfo_testing
+REPO=gcr.io/$(GCLOUD_PROJECT)
+release: REPO=nginmesh
+TAG=$(NGINMESH_VERSION)-$(GIT_COMMIT)
+TAG_LATEST=$(NGINMESH_VERSION)
+
+release:
```
Comment (sehz) [inline istio/tests/Makefile]: don't duplicate the make rule.  have a single rule but override using different target. please see agent makefile as example

File: istio/tests/cases/1_bd_spec.py [hunk 36]
```diff
@@ -0,0 +1,22 @@
+from __future__ import absolute_import
+import time
+import performance
+import configuration
+import logo
+from mamba import description, context, it
+from expects import expect, be_true, have_length, equal, be_a, have_property, be_none
+
+with description('nginmesh Test 01'):
```
Comment (sehz) [inline istio/tests/cases/1_bd_spec.py]: have better description of the test. say something like "testing basic functionality".  test 01 is not meaningful

**Comment by sehz** [comment]
use master as target not 0.6 branch

File: istio/tests/src/Dev.md [hunk 37]
```diff
@@ -0,0 +1,14 @@
+
```
Comment (sehz) [inline istio/tests/src/Dev.md]: remove this for now.  it is obsolete

File: istio/tests/cases/README.md [hunk 38]
```diff
@@ -0,0 +1,170 @@
+### Prerequisites
```
Comment (sehz) [inline istio/tests/cases/README.md]: move this file under tests not cases

Comment (sehz) [inline istio/tests/cases/README.md]: explain what this is. "This is to run end to end test suite...."

File: istio/tests/cases/install.sh [hunk 39]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+python -m pip install pipenv
```
Comment (sehz) [inline istio/tests/cases/install.sh]: don't install pipenv. let user install it. make as pre-requsitie

**Comment by sakows** [comment]
Changed the base branch from 0.6.0  to master

File: istio/tests/cases/README.md (hunk 38 above)
Comment (sakows) [reply istio/tests/cases/README.md]: ok

Comment (sakows) [reply istio/tests/cases/README.md]: ok

File: istio/tests/src/Dev.md (hunk 37 above)
Comment (sakows) [reply istio/tests/src/Dev.md]: ok

File: istio/release/samples/bookinfo/kube/route-rule-http-redirect.yaml (hunk 29 above)
Comment (sakows) [reply istio/release/samples/bookinfo/kube/route-rule-http-redirect.yaml:1]: as agreed to keep it.

File: istio/release/samples/bookinfo/kube/route-rule-all-v1.yaml (hunk 28 above)
Comment (sakows) [reply istio/release/samples/bookinfo/kube/route-rule-all-v1.yaml]: ok

File: istio/release/samples/bookinfo/kube/route-rule-http-redirect.yaml (hunk 29 above)
Comment (sakows) [reply istio/release/samples/bookinfo/kube/route-rule-http-redirect.yaml:1]: it should be updated, does not work otherwise.

File: istio/release/samples/bookinfo/kube/route-rule-http-retry.yaml (hunk 30 above)
Comment (sakows) [reply istio/release/samples/bookinfo/kube/route-rule-http-retry.yaml]: as agreed to keep it.

File: istio/release/samples/bookinfo/kube/route-rule-reviews-50-v3.yaml (hunk 31 above)
Comment (sakows) [reply istio/release/samples/bookinfo/kube/route-rule-reviews-50-v3.yaml]: ok

File: istio/release/samples/bookinfo/kube/route-rule-reviews-test-v2.yaml (hunk 32 above)
Comment (sakows) [reply istio/release/samples/bookinfo/kube/route-rule-reviews-test-v2.yaml]: ok

File: istio/tests/cases/1_bd_spec.py (hunk 36 above)
Comment (sakows) [reply istio/tests/cases/1_bd_spec.py]: ok

File: istio/tests/cases/install.sh (hunk 39 above)
Comment (sakows) [reply istio/tests/cases/install.sh]: ok

**Review by sehz — APPROVED** [review]
